### PR TITLE
Start saving `serotype` in the database and serving `serotype` on the API

### PR DESCRIPTION
**Pull request by SeanKennyNF**
Start saving `serotype` in the database and serving `serotype` on the API. Really tiny change but please feel free to send me any questions, I'd be happy to help clarify stuff about this API :)

### CODE REVIEW

File: src/api/arbo-resolvers.ts [hunk 1]
```diff
@@ -94,6 +97,7 @@ export const generateArboResolvers = (input: GenerateArboResolversInput): Genera
       pathogen,
       producer,
       sampleFrame,
+      serotype,
```
Comment (SeanKennyNF) [inline src/api/arbo-resolvers.ts:100]: This is the `serotype` that's served up by this endpoint.
```
query arbovirusFilterOptionsQuery {
  arbovirusFilterOptions {
    serotype
  }
}
```
```
data: {
  arbovirusFilterOptions: {
    serotype: [
      "DENV-1",
      "DENV-2",
      "DENV-3",
      "DENV-4",
      "E",
      "NS1"
    ]
  }
}
```

File: src/etl/types.ts [hunk 2]
```diff
@@ -22,6 +22,7 @@ export type AirtableEstimateFields = {
   "Sample Start Date": `${number}-${number}-${number}` | undefined;
   "Sample End Date": `${number}-${number}-${number}` | undefined;
   "Sample Size": number | undefined;
+  "Serotype": string[] | undefined;
```
Comment (SeanKennyNF) [inline src/etl/types.ts:25]: Key and type from Airtable

File: src/storage/types.ts [hunk 3]
```diff
@@ -12,6 +12,7 @@ export interface ArbovirusEstimateDocument {
   sampleFrame: string | undefined;
   sameFrameTargetGroup: string | undefined;
   sampleSize: number;
+  serotype: string[];
```
Comment (SeanKennyNF) [inline src/storage/types.ts:15]: Key and type in our database

File: src/etl/steps/transform-not-reported-values-to-undefined-step.ts [hunk 4]
```diff
@@ -31,23 +47,14 @@ export const transformNotReportedValuesToUndefinedStep = (
...
-                "nan",
-              ].includes(value)
-            ) {
+            if (isNotReportedValue(value)) {
               return [key, undefined];
             }
 
+            if(!!value && isArrayOfUnknownType(value)) {
+              return [key, value.filter((element) => !isNotReportedValue(element))]
+            }
```
Comment (EliasPinno) [inline src/etl/steps/transform-not-reported-values-to-undefined-step.ts:56]: `!!value` ?

File: src/etl/steps/transform-not-reported-values-to-undefined-step.ts [hunk 5]
```diff
@@ -19,6 +20,21 @@ interface TransformNotReportedValuesToUndefinedStepOutput {
...
 
+enum NotReportedValue {
+  NR = "NR",
+  nr = "nr",
+  "Not Reported" = "Not Reported",
+  "Not reported" = "Not reported",
+  "Not available" = "Not available",
+  "NA" = "NA",
+  "N/A" = "N/A",
+  "nan" = "nan",
```
Comment (EliasPinno) [inline src/etl/steps/transform-not-reported-values-to-undefined-step.ts:31]: Why are most of these `"quote" = "quote"` except the first two?

File: src/api/arbo-typedefs.ts [hunk 6]
```diff
@@ -26,6 +26,7 @@ export const arboTypedefs = `
     sampleSize: Int!
     sampleStartDate: String
     seroprevalence: Float
+    serotype: [String!]!
```
[the remark this answers is not in the document]
Comment (EliasPinno) [reply src/api/arbo-typedefs.ts:29]: Why is the bottom row invalid for `[String!]!`

**Review by EliasPinno — APPROVED** [review]

**Pull request merged**